### ISSUE TITLE
ubuntu 16.04 -> mysql 5.7 -> systemd startup error

### ISSUE THREAD

**Issue by aight8** [chef-cookbooks/mysql#440]
I think the problem is because of some misconfiguration of AppArmor. The mysqld process cannot open the **mysql-default/my.conf** file but the mysql user have access to it. I try now to disable this apparmor thing until this issue is not solved. Wondering why I'm the only one with this problem.
### Cookbook version

8
### Chef-client version

12
### Platform Details

ubuntu 16.04 xenial
### Steps to Reproduce:

```
mysql_service 'default' do
  version '5.7'
  package_version ''
  service_manager 'systemd'
  bind_address '0.0.0.0'
  port '3306'
  data_dir '/data/mysql'
  initial_root_password root_password
  action [:create, :start]
end
```
### Output

```
==> default: ================================================================================
==> default:     
==> default: Error executing action `start` on resource 'mysql_service[default]'
==> default:     
==> default: ================================================================================
==> default:     
==> default: 
==> default: 
==> default:     
==> default: Mixlib::ShellOut::ShellCommandFailed
==> default:     
==> default: ------------------------------------
==> default:     
==> default: mysql_service_manager_systemd[default] (/var/chef/cache/cookbooks/mysql/libraries/mysql_service.rb line 56) had an error: Mixlib::ShellOut::ShellCommandFailed: service[mysql-default] (/var/chef/cache/cookbooks/mysql/libraries/mysql_service_manager_systemd.rb line 99) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
==> default: 
==> default:     
==> default: ---- Begin output of /bin/systemctl --system start mysql-default ----
==> default: 
==> default:     
==> default: STDOUT: 
==> default: 
==> default:     
==> default: STDERR: Job for mysql-default.service failed because the control process exited with error code. See "systemctl status mysql-default.service" and "journalctl -xe" for details.
==> default: 
==> default:     
==> default: ---- End output of /bin/systemctl --system start mysql-default ----
==> default: 
==> default:     
==> default: Ran /bin/systemctl --system start mysql-default returned 1
```

**Comment by dmarkow** [comment]
Same issue here. On my previous server (cookbook version 6 and Ubuntu 14.04), starting a mysql server named "default", this cookbook created an `/etc/apparmor.d/local/mysql/default` that looked like this:

```
/etc/mysql-default/*.pem r,
/etc/mysql-default/conf.d/ r,
/etc/mysql-default/conf.d/* r,
/etc/mysql-default/my.cnf r,
/var/log/mysql-default/ r,
/var/log/mysql-default/* rw,
/data/mysql/ r,
/data/mysql/** rwk,
/run/mysql-default/** rw,
/run/mysql-default/mysqld.pid rw,
/run/mysql-default/mysqld.sock rw,
/tmp/mysql-default/ r,
/tmp/mysql-default/my.sql r,
```

Today, I updated to cookbook version 8 and deployed to Ubuntu 16.04. That same file now has this:

```
/etc//*.pem r,
/etc//conf.d/ r,
/etc//conf.d/* r,
/etc//my.cnf r,
/var/log/mysql-default/ r,
/var/log/mysql-default/* rw,
/data/mysql/ r,
/data/mysql/** rwk,
/run/mysql-default/** rw,
/run/mysql-default/mysqld.pid rw,
/run/mysql-default/mysqld.sock rw,
/tmp// r,
/tmp//my.sql r,
/* rw,
```

Looks like the `@mysql_name` variable isn't properly populating from the template?

**Comment by dmarkow** [comment]
It appears downgrading to v7.2.0 works fine for now.

**Comment by bassrock** [comment]
Experiencing the same issue as well

**Comment by cheeseplus** [comment]
This should have been fixed by #451 but if not please open a new issue with the latest version of the cookbook.
